### PR TITLE
#feature#11 login by verifying data in firebase

### PR DESCRIPTION
**Pull request by hancel-avila**
# Description
 - We've initialized the Firebase authenticator.
 - We've implemented the logInWithEmailAndPassword method to handle the login with user's email and password.
 
 Examples:
 - If the user already has an account, he will be able to log in on the Log in screen. Firebase will make the email and password validation.
 - Why?: These changes complete the user login and account creation experienece.
 
 
 # Testing
You can create a new user on the **Sign up** screen filling the info required (name, email & password) with valid credentials and press enter. 
Then click on the **Log in** text, down below the Sign Up with Google button, it will send you to the Log in screen.
Enter the previous email and password on the correspondant inputs and press enter.

image

### CODE REVIEW

File: FlightReservation/src/helpers/authState.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    const subscriber = auth().onAuthStateChanged(onAuthStateChanged);
+    return subscriber; // unsubscribe on unmount
+  }, []);
+
+  if (initializing) return null;
+
+  if (!user) {
+    return (
+      <View>
+        <Text>Login</Text>
```
Comment (DemisRincon) [inline FlightReservation/src/helpers/authState.js:26]: use the library I18n to put this messages

File: FlightReservation/src/helpers/authState.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      </View>
+    );
+  }
+
+  return (
+    <View>
+      <Text>Welcome {user.email}</Text>
```
Comment (DemisRincon) [inline FlightReservation/src/helpers/authState.js:33]: use the library I18n to put this messages

File: FlightReservation/src/screens/LogIn.js [hunk 3]
```diff
@@ -5,35 +5,40 @@ import {Formik} from 'formik';
...
     <SafeAreaView>
       {loading && <Text>Loading...</Text>}
+      {emailInUseError && <Text>Cannot log in</Text>}
+
       <Formik
         validationSchema={logInValidationSchema}
         initialValues={{
           email: 'juan1@example.com',
           password: 'Juan123%',
         }}
```
Comment (DemisRincon) [inline FlightReservation/src/screens/LogIn.js:39]: use data from inputs to test the function read the documentation to use formik

**Review by DemisRincon — CHANGES_REQUESTED** [review]
Check the comments

File: FlightReservation/src/screens/LogIn.js (hunk 3 above)
Comment (DemisRincon) [reply FlightReservation/src/screens/LogIn.js:39]: also you have merge conflicts pleas pull your main branch and solve the conflicts

**Review by DemisRincon — APPROVED** [review]

**Review by jkzasori — APPROVED** [review]

**Review by yovanha — APPROVED** [review]
Good job!

**Pull request merged**